salt/cloud/clouds/digital_ocean.py

```diff
@@ -694,17 +742,28 @@ def destroy(name, call=None):
     data = show_instance(name, call='action')
     node = query(method='droplets', droplet_id=data['id'], http_method='delete')
 
-    delete_record = config.get_cloud_config_value(
-        'delete_dns_record', get_configured_provider(), __opts__, search_global=False, default=None,
-    )
-
-    if delete_record and not isinstance(delete_record, bool):
+    ## This is all terribly optomistic:
+    # vm_ = get_vm_config(name=name)
+    # delete_dns_record = config.get_cloud_config_value(
+    #     'delete_dns_record', vm_, __opts__, search_global=False, default=None,
+    # )
+    # ToDo: when _vm config data can be made available, we should honor the configuration settings,
+    #         but until then, we should assume stale DNS records are bad, and default behavior should be to
+    #         delete them if we can.
+    delete_dns_record = True
+
+    if delete_dns_record and not isinstance(delete_dns_record, bool):
```

Comment:
This should be simplified to use `if not isinstance(delete_dns_record, bool):`` since `delete_dns_record` is already True.